### PR TITLE
llvm: restrict openmp ARM patch to stable version

### PR DESCRIPTION
**Pull request by chandlerc**
The relevant upstream patch was subsumed in:
https://reviews.llvm.org/D88252

This patch was landed 2020-11-24:
https://github.com/llvm/llvm-project/commit/7b5254223acbf2ef9cd278070c5a84ab278d7e5f

It had a bug that was fixed 2020-11-25:
https://github.com/llvm/llvm-project/commit/9e3e332d273b80b5167ac35f8dcfa7178e45c5e9

After these commits, the patch won't apply and isn't necessary.
Restricting it to the stable source fixes a build error with `--HEAD`.
I still can't get the `--HEAD` build to complete successfully on either
x86 or ARM because of an unrelated error during compilation, but this
gets much further than before.

Moved up above the livecheck to satisfy the `audit` results.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Review by SMillerDev — APPROVED** [review]

**Comment by chandlerc** [comment]
Thanks for the review! To be clear, I know nothing about Homebrew or the check failure. I'll leave it to you all to sort out and/or merge if appropriate. If there is something I can do, please let me know.

FYI, the only error I have with `--HEAD` builds of LLVM with Homebrew now is the following error, but I've not yet figured out the cause or how to debug it so I can't submit a PR to address it either. =/ But I know `--HEAD` isn't supported.
```
clang++: The build tool has reset ENV; --env=std required.
```

Anyways, thanks again!

**Comment by carlocab** [comment]
Does `brew install --HEAD --env=std llvm` work for you? It would be good to know if it does, since lldb is currently failing to build on 11.0.1: https://github.com/Homebrew/homebrew-core/pull/68393

**Pull request closed without merging**
